test/evp_extra_test.c

```diff
@@ -1094,6 +1153,80 @@ static int test_EC_priv_only_legacy(void)
 
     return ret;
 }
+
+static int test_evp_get_ec_pub_legacy(void)
+{
+    OSSL_LIB_CTX* libctx = NULL;
+    OSSL_LIB_CTX *oldctx = NULL;
+    unsigned char *pad = NULL;
+    EVP_PKEY *pkey = NULL;
+    EC_KEY *eckey = NULL;
+    BIGNUM *priv = NULL;
+    BIGNUM *x = NULL;
+    BIGNUM *y = NULL;
+    int ret = 0;
+
+    if (!TEST_ptr(libctx = OSSL_LIB_CTX_new()))
+        goto err;
+
+    if (!TEST_ptr(oldctx = OSSL_LIB_CTX_set0_default(libctx)))
```

Comment:
Can I ask why you need to make the call to set0_default() ?

Reply:
There is a EC_KEY_new_by_curve_name_ex()

Reply:
(Only set0_default() as a last resort... I see it as a hack)

Reply:
no reason really, will fix - I thought it was cleaner this way (the function you pointed too is not being used much) but sure, will fix